[FilterBridge] Various improvements

ORelio opened the pull request:
I'm using the Filter bridge to hide ads, sponsored articles and was missing a few features.
Here are a few improvements for the Filter bridge:

- Add option for case-insensitive regex
- Allow matching item content instead or in addition to item title
- Optionally attempt to convert encoding when applying matches

The changes should be backward-compatible with existing feeds based on the Filter bridge.
I tested the changes on my instance for a few days to check everything was working as expected.

ORelio committed:
[FilterBridge] Various improvements

- Add option for case-insensitive regex
- Allow matching item content instead or in addition to item title
- Optionally attempt to convert encoding when applying matches

ORelio committed:
Fix array append operator

ORelio commented:
(Please squash commits when merging)

em92 commented on bridges/FilterBridge.php:
```diff
@@ -2,80 +2,115 @@
+				'Title and Content' => 'title_content',
+			),
+			'defaultValue' => 'title',
+		),
```
I have case where author also should be filtered. If we add it `filter_target` values it will become messy.
Instead of this, I would suggest to make 3 checkboxes like `Apply filter to title` (default yes), `Apply filter to body` (default no), `Apply filter to author` (default no). In this case you also need to add check in collectData method

ORelio replied:
Good idea, doing that. However no need to change `collectData` because `parseItem` can hide an item by returning `null`.

ORelio committed:
Improve parameters

Use separate checkboxes for title, content, author
Change order in bridge card to make options clearer
Change max length to apply to all fields instead of only content

ORelio committed:
Fix array append operator (again)

em92 commented:
Hello, @captn3m0 !
This is what you asked https://github.com/RSS-Bridge/rss-bridge/issues/1960

em92 commented:
gj @ORelio !
